numba/transforms.py

```diff
@@ -61,12 +61,27 @@ def _loop_lift_get_candidate_infos(cfg, blocks, livemap):
         [callfrom] = loop.entries   # requirement checked earlier
         an_exit = next(iter(loop.exits))  # anyone of the exit block
         [(returnto, _)] = cfg.successors(an_exit)  # requirement checked earlier
+        inputs = livemap[callfrom]
+        outputs = livemap[returnto]
+
+        # ensure live variables are actually used in the blocks, else remove,
+        # saves having to create something valid to run through postproc
+        # to achieve similar
+        local_block_ids = set(loop.body) | set(loop.entries) | set(loop.exits)
+        used_vars = set()
+        for blk_id in local_block_ids:
+            blk = blocks[blk_id]
+            for inst in blk.body:
+                used_vars |= set([x.name for x in inst.list_vars()])
```

Comment:
This loop can be replaced with `numba.analysis.compute_use_defs`.

Sth like:

```

        from numba.analysis import compute_use_defs
        local_block_ids = set(loop.body) | set(loop.entries) | set(loop.exits)
        loopblocks = {}
        for k in local_block_ids:
            loopblocks[k] = blocks[k]

        used_vars = set()
        for vs in compute_use_defs(loopblocks).usemap.values():
            used_vars |= vs
```

Reply:
Thanks. That's a bit neater. Done in https://github.com/numba/numba/pull/2889/commits/8a31fe146678b8e7b471ba14e1990bbaa7db3e4d.